### ISSUE TITLE
Weird interaction between xonsh and pytest (Attribute Error XONSH_COLOR_STYLE)

### ISSUE THREAD

**Issue by con-f-use** [xonsh/xonsh#4592]
So I'm using pytest for a completely different project in it's on virtual environment, and got an xonsh problem:

```
python -m pytest --log-level=DEBUG --log-cli-level=DEBUG --show-capture=all -v -k test_ztd_support
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/tmp/qda_rest/.direnv/python-3.9.6/lib/python3.9/site-packages/_pytest/main.py", line 265, in wrap_session
INTERNALERROR>     config._do_configure()
INTERNALERROR>   File "/tmp/qda_rest/.direnv/python-3.9.6/lib/python3.9/site-packages/_pytest/config/__init__.py", line 982, in _do_configure
INTERNALERROR>     self.hook.pytest_configure.call_historic(kwargs=dict(config=self))
INTERNALERROR>   File "/tmp/qda_rest/.direnv/python-3.9.6/lib/python3.9/site-packages/pluggy/_hooks.py", line 277, in call_historic
INTERNALERROR>     res = self._hookexec(self.name, self.get_hookimpls(), kwargs, False)
INTERNALERROR>   File "/tmp/qda_rest/.direnv/python-3.9.6/lib/python3.9/site-packages/pluggy/_manager.py", line 80, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/tmp/qda_rest/.direnv/python-3.9.6/lib/python3.9/site-packages/pluggy/_callers.py", line 60, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/tmp/qda_rest/.direnv/python-3.9.6/lib/python3.9/site-packages/pluggy/_result.py", line 60, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/tmp/qda_rest/.direnv/python-3.9.6/lib/python3.9/site-packages/pluggy/_callers.py", line 39, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/nix/store/fh3chl81sk4l87bw06yakjmsbshhzx2r-xonsh-0.10.1/lib/python3.9/site-packages/xonsh/pytest_plugin.py", line 13, in pytest_configure
INTERNALERROR>     setup()
INTERNALERROR>   File "/nix/store/fh3chl81sk4l87bw06yakjmsbshhzx2r-xonsh-0.10.1/lib/python3.9/site-packages/xonsh/__amalgam__.py", line 21468, in setup
INTERNALERROR>     execer = Execer(xonsh_ctx=ctx)
INTERNALERROR>   File "/nix/store/fh3chl81sk4l87bw06yakjmsbshhzx2r-xonsh-0.10.1/lib/python3.9/site-packages/xonsh/__amalgam__.py", line 13967, in __init__
INTERNALERROR>     XSH.load(execer=self, ctx=xonsh_ctx)
INTERNALERROR>   File "/nix/store/fh3chl81sk4l87bw06yakjmsbshhzx2r-xonsh-0.10.1/lib/python3.9/site-packages/xonsh/built_ins.py", line 557, in load
INTERNALERROR>     self.env = kwargs.pop("env") if "env" in kwargs else Env(default_env())
INTERNALERROR>   File "/nix/store/fh3chl81sk4l87bw06yakjmsbshhzx2r-xonsh-0.10.1/lib/python3.9/site-packages/xonsh/__amalgam__.py", line 18351, in __init__
INTERNALERROR>     self[key] = val
INTERNALERROR>   File "/nix/store/fh3chl81sk4l87bw06yakjmsbshhzx2r-xonsh-0.10.1/lib/python3.9/site-packages/xonsh/__amalgam__.py", line 18585, in __setitem__
INTERNALERROR>     deval = detyper(val)
INTERNALERROR>   File "/nix/store/fh3chl81sk4l87bw06yakjmsbshhzx2r-xonsh-0.10.1/lib/python3.9/site-packages/xonsh/__amalgam__.py", line 5848, in detype
INTERNALERROR>     return x.detype()
INTERNALERROR>   File "/nix/store/fh3chl81sk4l87bw06yakjmsbshhzx2r-xonsh-0.10.1/lib/python3.9/site-packages/xonsh/__amalgam__.py", line 16983, in detype
INTERNALERROR>     style = self.style
INTERNALERROR>   File "/nix/store/fh3chl81sk4l87bw06yakjmsbshhzx2r-xonsh-0.10.1/lib/python3.9/site-packages/xonsh/__amalgam__.py", line 17015, in style
INTERNALERROR>     style_name = self.style_name
INTERNALERROR>   File "/nix/store/fh3chl81sk4l87bw06yakjmsbshhzx2r-xonsh-0.10.1/lib/python3.9/site-packages/xonsh/__amalgam__.py", line 17006, in style_name
INTERNALERROR>     env_style_name = env.get("XONSH_COLOR_STYLE", "default")
```

Guess it has something to do with pytest plugins and xonsh. I'm at a loss and don't have time to debug.

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by anki-code** [comment]
> XONSH_COLOR_STYLE

Hi @dyuri! Could you please take a look. May be it's related to something around #4253

**Comment by dyuri** [comment]
@con-f-use could you add a small example script that I could use to reproduce this issue?

**Comment by gforsyth** [comment]
This is fixed on 0.11.0 -- can we update the nix pkg in question?

**Comment by con-f-use** [comment]
It's [already updated](https://github.com/NixOS/nixpkgs/blob/master/pkgs/shells/xonsh/default.nix), just didn't make it into [stable](https://github.com/NixOS/nixpkgs/blob/nixos-21.11/pkgs/shells/xonsh/default.nix) by a hair's width.

**Comment by gforsyth** [comment]
Fixed in #4438, I believe.  Some command caching behavior sometimes led (almost always at startup or when running test fixture) to `__xonsh__.env` remaining uninitialized, so the `get` call didn't work.  
If you cherry pick #4438 that should fix the crash
